modules/ray_tracing/include/ray_kernels/RayKernel.h

```diff
@@ -0,0 +1,111 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// Local Includes
+#include "IntegralRayKernelBase.h"
+
+// MOOSE Includes
+#include "MooseVariableInterface.h"
+#include "TaggingInterface.h"
+
+// Forward declarations
+template <typename>
+class RayKernelTempl;
+
+using RayKernel = RayKernelTempl<Real>;
+// Not implementing this until there is a use case and tests for it!
+// using VectorRayKernel = RayKernelTempl<RealVectorValue>;
+
+/**
+ * Base class for a ray kernel that contributes to the residual and/or Jacobian
+ */
+template <typename T>
+class RayKernelTempl : public IntegralRayKernelBase,
+                       public MooseVariableInterface<T>,
+                       public TaggingInterface
+{
+public:
+  RayKernelTempl(const InputParameters & params);
+
+  static InputParameters validParams();
+
+  /**
+   * The MooseVariable this RayKernel contributes to
+   */
+  MooseVariableFE<T> & variable() { return _var; }
```

Comment:
Could we make this a `MooseVariableField<T>`?  Then we can try to ensure this works with FV as well as FE. - same with the member variable _var.  This may need to be addressed in other (BCs, ADRayKernel, etc.) classes as well.

Reply:
I was going to say this but I almost would rather we leave it there so that we know no one can try MOC with FV for now. Better to have it broken for a simple reason than risk it being half-working but actually not.

Either that or add a log/check for FV